### PR TITLE
1058-enrichWithCulturegraphRvkWithFix

### PR DESCRIPTION
**Pull request by TobiasNx**
See #1058

I tried to update the draft from https://github.com/hbz/lobid-resources/tree/1058-enrichWithCulturegraphRvk to a version working with fix.
I merged the master into this branch.

In contrast to the morph approach:
- We do not need a separate filter step since the fix already can do this.

Also I do not know if we need:

https://github.com/hbz/lobid-resources/blob/d12b7a34a29ef21c282b0040d74cdb1390345129/src/main/java/org/lobid/resources/run/CulturegraphXmlFilterHbzToJson.java#L30-L45

The decode-marcXml mechanism correctly identified single records.

### CODE REVIEW

**Comment by TobiasNx** [comment]
Somehow it does not find the fix file yet.

**Comment by TobiasNx** [comment]
Additionally we would need zdbId as mapping parameter, because new zdb resources dont get hbzIds anymore.

**Comment by dr0i** [comment]
> * `id` is always in the first column

a) As said in https://github.com/hbz/lobid-resources/pull/1921#issuecomment-2135449004 this should be done and makes only sense when creating a CSV. If it's not possible using _one_ FIX  could you provide a second one?
 
b) > all `id's` have their rows
Hm, should'nt that be possible with using triples somehow ? Anyway. Then we shoud make the first ID to the only ID (the only value for column one).

**Comment by dr0i** [comment]
The build of the concordance just started, based on the 9,2 GB file  `aggregate_20240507.marcxml.gz` mentioned in https://github.com/hbz/lobid-resources/issues/1058#issue-572148389.
At a first glance - there are many doublette RVK entries, e.g.:
> "990191558260206441","HU 8423, HU 8424, HU 8423, HU 8424, HU 8424, HU 8423, HU 8424"

Can you prevent these doublettes via the FIX @TobiasNx ?

File: src/main/resources/rvk/cg-to-rvk-json.fix
Comment (TobiasNx) [inline src/main/resources/rvk/cg-to-rvk-json.fix:1]: Why do we need two fixes?

Comment (dr0i) [reply src/main/resources/rvk/cg-to-rvk-json.fix:1]: `json` doesn't make a record for every `id` - the _search_ is done by the search engine, in contrast to a csv where we nee a single unique key. The `json` is moe performant when using search engines, the `csv` is the only way to go when using tables.

File: src/main/resources/rvk/cg-to-rvk-json.fix [hunk 1]
```diff
@@ -0,0 +1,29 @@
+set_array("rvk[]")
+
+do list(path: "084??", "var": "$i")
+    if any_match("$i.2", "rvk")
+        copy_field("$i.a","rvk[].$append")
+    end
+end
+
```
Comment (TobiasNx) [inline src/main/resources/rvk/cg-to-rvk-json.fix:8]:
```suggestion
uniq("rvk[]")
```

**Comment by dr0i** [comment]
I am going to merge this. 
Note that the branch name is not good, resp. that a new PR is going to be made where we not only generate the concordance but enrich our lobid-resources data with it.

**Review by dr0i — APPROVED** [review]

**Pull request merged**